thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -268,6 +269,19 @@ namespace diskann {
     }
     this->count_visited_nodes.store(true);
 
+    // sync allocate memory
+    if (nhood_cache_buf == nullptr) {
+        nhood_cache_buf = new unsigned[num_nodes_to_cache * (max_degree + 1)];
+        memset(nhood_cache_buf, 0, num_nodes_to_cache * (max_degree + 1));
```

Comment:
`sizeof(unsigned)` is 4 on my machine, so memset should include `sizeof(unsigned)` as well.
But instead I would just  convert `nhood_cache_buf` from a pointer to a `std::vector<>`.